### PR TITLE
fix: email statuses

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Email wasn't respecting the statuses and as a result it was returning both success and error at the same time.

Reason for that was that we didn't take into account the event status strings. 

This PR fixes that.

Need to thouroughly test it out to understand when it works and when it doesn't ~ I think there are some issues on iOS 17 for example wasn't able to see the actual html template.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/screens/Artwork/useMailComposer.ts [hunk 1]
```diff
@@ -38,31 +100,14 @@ export const useMailComposer = () => {
...
-
-      if (!mailError) {
-        toast.show({
-          title: "Email sent.",
-          type: "info",
-        })
-      }
-
-      mailError = ""
+      emailComposer({
```
Comment (damassi) [inline src/screens/Artwork/useMailComposer.ts:103]: Ok that this isn't `await`'d?

**Review by damassi — APPROVED** [review]
One q, but otherwise tyyy

File: src/screens/Artwork/useMailComposer.ts [hunk 2]
```diff
@@ -8,6 +8,68 @@ import { GlobalStore } from "system/store/GlobalStore"
...
+
+const emailComposer = async ({
+  subject,
+  ccRecipients,
+  body,
+  isHTML,
+  attachments,
+  toast,
+}: EmailComposerProps) => {
+  await Mailer.mail(
```
Comment (gkartalis) [inline src/screens/Artwork/useMailComposer.ts:28]:
```suggestion
  Mailer.mail(
```

File: src/screens/Artwork/useMailComposer.ts [hunk 3]
```diff
@@ -8,6 +8,68 @@ import { GlobalStore } from "system/store/GlobalStore"
...
+interface EmailComposerProps {
+  subject: string
+  ccRecipients?: string[]
+  body: string
+  isHTML: boolean
+  attachments?: any[]
+  toast: ReturnType<typeof useToast>["toast"]
+}
+
+const emailComposer = async ({
```
Comment (gkartalis) [inline src/screens/Artwork/useMailComposer.ts:20]:
```suggestion
const emailComposer = ({
```

File: src/screens/Artwork/useMailComposer.ts (hunk 1 above)
Comment (gkartalis) [reply src/screens/Artwork/useMailComposer.ts:103]: no need to await that, will also remove the asyn await from the emailComposer func

**Pull request merged**